addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyScanRule.java

```diff
@@ -313,54 +302,89 @@ private String getCSPNoticesString(ArrayList<Notice> notices) {
         return matchedHeaders;
     }
 
-    private List<String> getAllowedWildcardSources(String policyText, Origin origin) {
+    private List<String> getAllowedWildcardSources(String policyText) {
 
         List<String> allowedSources = new ArrayList<>();
-        Policy pol = ParserWithLocation.parse(policyText, origin);
-
-        if (pol.allowsScriptFromSource(PARSED_WILDCARD_URI)) {
-            allowedSources.add("script-src");
-            allowedSources.add("script-src-elem");
-            allowedSources.add("script-src-attr");
-        }
-        if (pol.allowsStyleFromSource(PARSED_WILDCARD_URI)) {
-            allowedSources.add("style-src");
-            allowedSources.add("style-src-elem");
-            allowedSources.add("style-src-attr");
-        }
-        if (pol.allowsImgFromSource(PARSED_WILDCARD_URI)) {
-            allowedSources.add("img-src");
-        }
-        if (pol.allowsConnectTo(PARSED_WILDCARD_URI)) {
-            allowedSources.add("connect-src");
-        }
-        if (pol.allowsFrameFromSource(PARSED_WILDCARD_URI)) {
-            allowedSources.add("frame-src");
-        }
-        if (pol.allowsFrameAncestor(PARSED_WILDCARD_URI)) {
-            allowedSources.add("frame-ancestors");
-        }
-        if (pol.allowsFontFromSource(PARSED_WILDCARD_URI)) {
-            allowedSources.add("font-src");
-        }
-        if (pol.allowsMediaFromSource(PARSED_WILDCARD_URI)) {
-            allowedSources.add("media-src");
-        }
-        if (pol.allowsObjectFromSource(PARSED_WILDCARD_URI)) {
-            allowedSources.add("object-src");
-        }
-        if (pol.allowsManifestFromSource(PARSED_WILDCARD_URI)) {
-            allowedSources.add("manifest-src");
-        }
-        if (pol.allowsWorkerFromSource(PARSED_WILDCARD_URI)) {
-            allowedSources.add("worker-src");
-        }
-        if (pol.allowsPrefetchFromSource(PARSED_WILDCARD_URI)) {
-            allowedSources.add("prefetch-src");
-        }
-        if (pol.allowsFormAction(PARSED_WILDCARD_URI)) {
-            allowedSources.add("form-action");
+        Policy pol = Policy.parseSerializedCSP(policyText, PolicyErrorConsumer.ignored);
+        Optional<URI> httpUri = URI.parseURI("http://" + RAND_FQDN);
+        Optional<URI> httpsUri = URI.parseURI("https://" + RAND_FQDN);
+
+        if (httpUri.isPresent() && httpsUri.isPresent()) {
```

Comment:
SonarLint insisted these be checked. I think it's kinda over-kill given that the value is being set directly/specifically, but maybe I'm missing some edge case about `Optional`s